aws_batch/pipeline.py

```diff
@@ -90,6 +90,18 @@
 LOGS_OUT = 'log.txt'
 
 ### convenience functions
+def count_reads(file_name, file_type):
+    if file_type == "fastq_gz_paired":
+        return 2*sum(1 for line in gzip.open(file_name))
```

Comment:
this is a very inefficient way of counting lines. basically you are creating a temporary array and sum them together. 
If the fastq file has millions of lines, you are using millions of bytes of memory just for this. 
I think you can just start with count=0 and count+=1 

Actually I will redo the function this way. 
count =0 
if file_name[-3:] == '.gz': 
  infile = gz.open(file_name)
else:
  infile = open(filename)

for line in inline:
  if file_type == 'fastq':
    count += 2./4
 elif file_type == 'fasta':
    count += 1
  elif file_type == 'fasta_paired':
    count += 2 if line.startswith(('>')
  else:
     count += 1 
infile.close()
 return count

Reply:
Okay, just make sure to use file_type "fastq_paired" because the run_and_log function checks if "paired" occurs in file_type to determine messages